### DISCUSSION TITLE
next-themes Error: (0 , react__WEBPACK_IMPORTED_MODULE_0__.createContext) is not a function

### DISCUSSION THREAD

**Discussion by NotNotEvan** [vercel/next.js#59892] [Help]
### Summary

I'm receiving this error after installing next-themes and I'm unable to locate the exact issue.
![localhost_3000_](https://github.com/vercel/next.js/assets/110955018/29bc7c1d-3bc3-4f44-8be3-c207912dd319)
![localhost_3000_ (1)](https://github.com/vercel/next.js/assets/110955018/e668e12e-1370-4678-a939-31c3f3abf665)

I've tried deleting the node_modules folder and package-lock.json, then running `npm install`. This did not work.

I've tried clearing my next cache by deleting the .next folder and running `npm run dev` again. This also did not work.

I've seen online that I should insert `"use client"` at the top of my file, which I have done. But that also has not solved this issue.

I've completely narrowed this issue down to next-themes.... now what are my next-steps... see what I did there ;)... stupid.

Anyways, I'm not sure what else to check at this point, but any help is GREATLY appreciated. Thank you!

**Comment by DevendraBhoraniya** [comment]
I also had the same problem your only option is to create a ThemeClient component and then use that instead of the theme provider 

here are screen shots 

![Screenshot 2024-01-28 144940](https://github.com/vercel/next.js/assets/94428451/e26c9076-b004-465a-bf86-d48b0f4c6631)

![Screenshot 2024-01-28 145025](https://github.com/vercel/next.js/assets/94428451/d64c478f-5776-4875-88a6-77066fc34c64)

**Comment by rjfranc0** [comment]
I'm facing the same issue, I already wraped the Provider from `next-themes` into a custom component with `'use client'` but nothing seems to work. Please someone from nextjs or next-themes answer this and everything @NotNotEvan did above but still.

🙏🏾🙏🏾🙏🏾

**Reply by DevTK0** [reply]
Try importing it under a different name.

```
import { ThemeProvider as NextThemeProvider } from "next-themes";

...
<NextThemeProvider> {children} </NextThemeProvider>
```

**Comment by JakeerC** [comment]
add `'use client'` at top of the component which uses `useTheme` from `next-themes`.

**Reply by luiznegreiros** [reply]
It is not good...
With this, every component bellow will be a 'use cliente' component.

There's any other alternative to this issue?

**Reply by JakeerC** [reply]
That's big misconception, 
Only imports of 'use client' component will be part of bundle and components may be in client side rendering everything else will be server side

Refer below links for more information 
https://github.com/reactwg/server-components/discussions/4

https://www.joshwcomeau.com/react/server-components/

**Reply by LeoJunioYuri** [reply]
It worked for me. Thank you very much!

**Reply by suprunchuk** [reply]
THX

**Reply by brytebee** [reply]
It worked for m as well, thanks.

**Comment by krisnaw** [comment]
any update on this? Having a similar error when using the Sonner component.

**Comment by abhishek1savaliya** [comment]
'use client'
import React from 'react';
import { Progress } from "@material-tailwind/react";

const Page = () => {
    return (
        <div >
            <h1>hello</h1>
            <Progress value={50} />
            <h1>hello</h1>
        </div>
    )
}

export default Page;

  <Progress value={50} /> this component is not working.

**Comment by PassertlonG** [comment]
Hey guys, I also encountered this problem just now. Here is the solution, and then I will tell you how I discovered this solution.
**
You should add "use client" at the top of the file  named"page. tsx" (or "page. jsx")
**
 [https://nextui.org/docs/customization/dark-mode#add-the-theme-switcher](url)
 "use client" at top of file named "page.tsx"

**Reply by PassertlonG** [reply]
@NotNotEvan @krisnaw @luiznegreiros @FrancoRATOVOSON @DevendraBhoraniya

**Reply by Nazarinejad** [reply]
i've been looking for a solution for 3 hours!!!
thank youuuuuuu

**Comment by sachinwak** [comment]
use `<html lang="en" suppressHydrationWarning>`  in RootLayout. It work for me
